### PR TITLE
Enhance the define_app decorator

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

File: src/cogent3/app/io_new.py [hunk 1]
```diff
@@ -432,9 +432,12 @@ def main(self, /, data: SerialisableType, *, identifier=None) -> T:
         identifier
         """
         identifier = identifier or get_data_source(data)
```
Comment (Nick-Foto) [inline src/cogent3/app/io_new.py:434]: Here we should use `get_data_source`  or new `open_data_source` ?

Comment (GavinHuttley) [reply src/cogent3/app/io_new.py:434]: Will move to `open_data_source` when that is merged in

File: src/cogent3/app/composable.py [hunk 2]
```diff
@@ -1075,7 +1075,9 @@ def _main(self, arg, *args, **kwargs):
     return result
 
 
-def define_app(klass=None, *, app_type: AppType = GENERIC):
+def define_app(
+    klass=None, *, app_type: AppType = GENERIC, skip_not_completed: bool = True
+):
     """decorator for building callable apps
```
Comment (Nick-Foto) [inline src/cogent3/app/composable.py:1081]: I think default value for `skip_not_completed` should be False. Because I think most cases is False,

Comment (GavinHuttley) [reply src/cogent3/app/composable.py:1081]: I interpret `skip_not_completed=True` to mean that NotCompleted's will not be processed, which is the most common use case. A developer has to take an action of setting it to False for it to change the existing behaviour.

That said, if you have a suggestion for a different name, I'm happy to hear it!

Comment (Nick-Foto) [reply src/cogent3/app/composable.py:1081]: Its name is good, I thought default is to not skip.

File: src/cogent3/app/io_new.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/app/io_new.py:561]: Actually, read that closer now. It's not `get_data_source` not **store**.

Comment (GavinHuttley) [reply src/cogent3/app/io_new.py:561]: I get confused by those too! But we have `open_data_store` I won't get confused.

**Pull request merged**